src/System.Net.Http/src/System/Net/Http/Unix/CurlException.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+
+namespace System.Net.Http
+{
+    internal sealed class CurlException : Exception
+    {
+        internal CurlException(int error, string message) : base(message)
+        {
+            HResult = error;
+        }
+
+        internal CurlException(int error, bool isMulti) : base(GetCurlErrorString(error, isMulti))
+        {
+            HResult = error;
```

Comment:
Instead of storing the HResult directly in both ctors, you might just change this ctor to delegate to the other, e.g.
```C#
internal CurlException(int error, bool isMulti) : 
    base(error, GetCurlErrorString(error, isMulti))
{
}
```